gateway/src/events/Message.ts

```diff
@@ -1,32 +1,47 @@
-import { CLOSECODES, OPCODES } from "../util/Constants";
+import { CLOSECODES } from "../util/Constants";
 import { WebSocket, Payload } from "@fosscord/gateway";
-var erlpack: any;
+let erlpack: any;
 try {
 	erlpack = require("@yukikaze-bot/erlpack");
 } catch (error) {}
 import OPCodeHandlers from "../opcodes";
-import { Tuple } from "lambert-server";
 import { check } from "../opcodes/instanceOf";
 import WS from "ws";
 
 const PayloadSchema = {
 	op: Number,
-	$d: new Tuple(Object, Number), // or number for heartbeat sequence
+	$d: Object || Number, // or number for heartbeat sequence
 	$s: Number,
 	$t: String,
 };
 
-export async function Message(this: WebSocket, buffer: WS.Data) {
+export async function Message(this: WebSocket, buffer: WS.RawData) {
 	// TODO: compression
-	var data: Payload;
+	let data: Payload;
 
 	if (this.encoding === "etf" && buffer instanceof Buffer)
 		data = erlpack.unpack(buffer);
-	else if (this.encoding === "json" && typeof buffer === "string")
-		data = JSON.parse(buffer);
-	else return;
+	else if (this.encoding === "json")
+		data = JSON.parse(buffer as unknown as string); //TODO: is this even correct?? seems to work for web clients...
+	else if(/--debug|--inspect/.test(process.execArgv.join(' '))) {
+		debugger;
+		return;
+	}
+	else {
+		console.log("Invalid gateway connection! Use a debugger to inspect!");
+		return;
+	}
 
-	check.call(this, PayloadSchema, data);
+	if(process.env.WS_VERBOSE)
+		console.log(`[Websocket] Incomming message: ${JSON.stringify(data)}`);
+	if(data.op !== 1)
+		check.call(this, PayloadSchema, data);
+	else { //custom validation for numbers, because heartbeat
+		if(data.s || data.t || typeof data.d !== "number") {
+			console.log("Invalid heartbeat...");
+			this.close(CLOSECODES.Decode_error);
```

Comment:
This isn't an issue on Slowcord, or when I test master branch locally. I wonder what broke?

Reply:
Type checks broke, likely due to newer typescript resolving `Object || Number` to `Object`...